common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -541,24 +540,32 @@ public void copy(int srcIdx, char[] dst, int dstIdx, int length) {
             throw new NullPointerException("dst");
         }
 
-        final int thisLen = value.length;
-
-        if (srcIdx < 0 || length > thisLen - srcIdx) {
+        if (srcIdx < 0 || length > length() - srcIdx) {
             throw new IndexOutOfBoundsException("expected: " + "0 <= srcIdx(" + srcIdx + ") <= srcIdx + length("
-                            + length + ") <= srcLen(" + thisLen + ')');
+                            + length + ") <= srcLen(" + length() + ')');
         }
 
         final int dstEnd = dstIdx + length;
-        for (int i = srcIdx, j = dstIdx; j < dstEnd; i++, j++) {
-            dst[j] = b2c(value[i]);
+        for (int i = dstIdx, j = srcIdx + arrayOffset(); i < dstEnd; i++, j++) {
+            dst[i] = b2c(value[j]);
         }
     }
 
     @Override
+    public AsciiString subSequence(int start) {
```

Comment:
Humm...I think so. If this method is called the type must be of type `AsciiString`, and the underlying `subSequence(..)` implementation should be fine because it is just working on the byte array bcz of the reduced charset.  Am I missing something?